pkgs/development/tools/java/jprofiler/default.nix

```
@@ -1,45 +1,45 @@
 { stdenv
 , lib
+, callPackages
 , fetchurl
 , makeWrapper
 , makeDesktopItem
 , copyDesktopItems
-, undmg
+, autoPatchelfHook
+, callPackage
 , jdk
+, libX11
+, musl
+, component ? "ui"
 }:
 
 let
-  inherit (stdenv.hostPlatform) system;
-  pname = "jprofiler";
+  versions = import ./version.nix;
 
-  # 11.1.4 is the last version which can be unpacked by undmg
-  # See: https://github.com/matthewbauer/undmg/issues/9
-  version = if stdenv.isLinux then "13.0.2" else "11.1.4";
   nameApp = "JProfiler";
 
+in stdenv.mkDerivation (finalAttrs: rec {
```

Pretty sure this can be removed now.
```suggestion
in stdenv.mkDerivation rec {
```

---

There's no need to use it there. Take a look at [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/faircamp/default.nix#L33).

---

@azahi I feel that I've missed how can I define used packet (`jprofiler` or `jprofiler-ui`) inside test to use the right path.

---

Oh, good point. Okay then.

---

@azahi let me squash everything into one commit when